ocelli-rxnetty/src/main/java/netflix/ocelli/rxnetty/FailureListener.java

```diff
@@ -0,0 +1,38 @@
+package netflix.ocelli.rxnetty;
+
+import rx.Scheduler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A contract for taking actions upon detecting an unhealthy host. A failure listener instance is always associated with
+ * a unique host, so any action taken on this listener will directly be applied to the associated host.
+ */
+public interface FailureListener {
```

Comment:
I think decoupling the quarantine strategy from the client(listener in this case) makes things more complex than adds value. Separating the quarantine strategy has these issues:

- Logically, the failure detection decision would be made by listening to events from the client and hence they are tied to the client. With this separation, one has to understand how and why to publish events to ocelli with eventually the only value add being to quarantine (notice that this listener is maintaining the metrics himself). Why to go around this hoop if you can just make the quarantine decision locally?
- Concepts of removing or quarantining the hosts are not easily discoverable as they are hidden behind an event abstraction. 
- The client (listener) will have to split the logic of maintaining metrics and the strategies for quarantine and removal in different abstractions, when they are logically makes sense together (take actions based on metrics). This leads to less intuitive code.

Based on this work of integrating RxNetty with ocelli, one thing which is evident to me is that all actions (failure detection, changing weights of hosts, etc.) would originate from listening to events from a client and these actions are discrete and synchronous, so providing abstractions to take these actions indirectly adds more complexity than convenience. Instead, if we have low level constructs(read functions) that aids making these decisions are better than creating, may I say a framework, for taking these actions.